pkg/clustersserver/clustersserver.go

```diff
@@ -28,14 +29,32 @@ type Server struct {
 	ClientCache       clientCache
 	AvailableContexts []string
 	InitialContext    string
+	CreateClient      func(string) (client.Client, error)
+}
+
+func NewServer(kubeContexts []string, currentKubeContext string) http.Handler {
+	clusters := Server{
+		AvailableContexts: kubeContexts,
+		InitialContext:    currentKubeContext,
+		ClientCache:       map[string]client.Client{},
+		CreateClient:      defaultCreateClient,
+	}
+
+	clustersHandler := pb.NewClustersServer(&clusters, nil)
+
+	return clustersHandler
+}
+
+func defaultCreateClient(kubeContext string) (client.Client, error) {
+	return util.NewKubeClient(kubeContext)
 }
 
 func (s *Server) getClient(kubeContext string) (client.Client, error) {
 	if s.ClientCache[kubeContext] != nil {
```

Comment:
This looks like it's going to have concurrent access. I'd guard it with a mutex.

Reply:
Should be resolved now.